### DISCUSSION TITLE
Welcome to ROMP Discussions!

### DISCUSSION THREAD

**Discussion by Arthur151** [Arthur151/ROMP#16] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by FishWoWater** [comment]
Hi!  I am wondering about the method to learn global depth ordering. Did you try to learn a translation vector for each person and a single camera for the image? (i.e. 3N + 3 parameters, in contrast to current 3N parameters from camera map where N is the number of people) Actually SMPL encodes a prior and I think this method should work? @Arthur151

**Reply by Arthur151** [reply]
Global depth ordering is intereting but challenge. It could be easier to estimate the depth order index (as you mensioned before). If we try to retrieve the actual camera metrix, it could be a little bit harder to get accurate parameters. But it should be work with proper settings.

**Reply by FishWoWater** [reply]
> Global depth ordering is intereting but challenge. It could be easier to estimate the depth order index (as you mensioned before). If we try to retrieve the actual camera metrix, it could be a little bit harder to get accurate parameters. But it should be work with proper settings.

Alright, I am gonna try this with your training code. Looking forward to 🥇 :)

**Reply by Arthur151** [reply]
Taining code has been released and tested by developers.

**Comment by AlexKTracerMain** [comment]
Hey Arthur! I saw this posted in the CVPR2022 booted up a demo in under 30min today, very impressive! Think any performance improvements possible if we're able to provide known camera parameters? 

Also, how to optimize to run on GPU (instead of just CPU)? OR even to use all my cores on CPU. Great work! First image segmentation I've seen that comes close to what my brain is able to infer from a single monocular.

**Reply by Arthur151** [reply]
Hi, Alex,
Sure, if we are able to get the camera intrinsic | extrinsic parameters during training. I guess the accuracy of both 3D pose & 3D localization would be improved. Because it solves the mis-alignment between supervision caused by our wrong predefined camera parameters. 

Maybe deploy the model with ONNX / TensorRT would be helpful.

**Comment by zhangby2085** [comment]
Hey Arthur! Great job for the pose extraction, excellent task in visualization of the 3d locations, a question to ask, by giving the function: romp --mode=webcam --show , is that possible to change the character and the background, the romp -h for help, does not show relevant information?

**Reply by Arthur151** [reply]
I think this is what you need.

https://github.com/yanch2116/CDBA

**Reply by zhangby2085** [reply]
Thank you very much, Arthur, thumbs up, I tried the above repo, and it works, it is a pity that it does not apply the --show_largest function, for example, if there are multiple in front of the camera, the Armature is keeping on jumping from one to another.

**Comment by WUchuhan077** [comment]
Hi,Arthur.During my inference for trace.Something wrong occoured. PermissionError: [Errno 13] Permission denied: '/home/yusun' 
I don't know how to solve it.

**Comment by gregcockroft** [comment]
There are some hardcoded paths inside of this project. Search for '/home/yusun' in all the python code. Replace it with your home directory.

**Comment by WUchuhan077** [comment]
Thank you!
Alright done.Really good.



发自我的iPhone


------------------ Original ------------------
From: Greg Cockroft ***@***.***&gt;
Date: Tue,Jun 25,2024 7:43 PM
To: Arthur151/ROMP ***@***.***&gt;
Cc: WUchuhan077 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [Arthur151/ROMP] Welcome to ROMP Discussions! (Discussion #16)





 
There are some hardcoded paths inside of this project. Search for '/home/yusun' in all the python code. Replace it with your home directory.

**Comment by WUchuhan077** [comment]
Hi,Arthur.When I downloaded the h36m files from Baidu network, I noticed that these are some files missing. How could I download the missing files?

**Reply by WUchuhan077** [reply]
There are only images.z02 03 04 05 06 07. The others are missing.
